### ISSUE TITLE
Fix false negatives for level 3 not() psuedo-classes in selector-max-specificity

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#2870]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A false negatives bug for [level 3 `:not()` psuedo-classes](https://www.w3.org/TR/css3-selectors/#negation). The selector [is ignored outright](https://github.com/stylelint/stylelint/blob/master/lib/rules/selector-max-specificity/index.js#L46), when it should only be ignored if it's a level 4 `:not()` selector i.e. contains anything other than a _simple selector_.

Ref: https://github.com/stylelint/stylelint/issues/2857

> Which rule, if any, is this issue related to?

`selector-max-specificity`

> What CSS is needed to reproduce this issue?

```css
a b:not(a) {}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "selector-max-specificity": "0,0,2"
  }
}
```

> Which version of stylelint are you using?

`8.1.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Demo

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Nope, valid Level 3 Selector.

> What did you expect to happen?

The following warning:

```shell
1:1 Expected "a b:not(c)" to have a specificity no more than "0,0,2"
```

> What actually happened (e.g. what warnings or errors you are getting)?

No warning

**Comment by jeddy3** [comment]
No longer relevant due to approach in #2857
